agents/check_mk_agent.openwrt

```diff
@@ -336,9 +336,10 @@ fi
 # Current state of bonding interfaces
 if [ -e /proc/net/bonding ] ; then
     echo '<<<lnx_bonding:sep(58)>>>'
-    pushd /proc/net/bonding > /dev/null
+    olddir=$(pwd)
```

Comment:
Hi,
Have you considered/tried a subshell instead? e.g.

    # Current state of bonding interfaces
    if [ -e /proc/net/bonding ]; then
        echo '<<<lnx_bonding:sep(58)>>>'
        (
            cd /proc/net/bonding || return
            head -v -n 1000 ./*
        )
    fi

Note that this example code (that I wrote over a year ago) also corrects two other issues that Shellcheck raises

* https://github.com/koalaman/shellcheck/wiki/SC2164
* https://github.com/koalaman/shellcheck/wiki/SC2035

Reply:
You are right, this would also work.